archive/archive_ingest/src/archive_ingest.py

```diff
@@ -0,0 +1,68 @@
+# -*- encoding: utf-8 -*-
+"""
+Receives a message to ingest a bag giving the URL and publishes the archive event to an SNS topic.
+"""
+
+import os
+
+import boto3
+import daiquiri
+from urllib.parse import urlparse
+
+from wellcome_aws_utils.sns_utils import publish_sns_message
+from wellcome_aws_utils.lambda_utils import log_on_error
+
+daiquiri.setup(level=os.environ.get('LOG_LEVEL', 'INFO'))
+logger = daiquiri.getLogger()
+
+
+def ensure_valid_request(event):
+    required_key = 'bagURL'
+    if type(event) is not dict:
+        raise ValueError(f"Invalid request not json dict: {event}")
+    if required_key not in event:
+        raise ValueError(f"Invalid request missing {required_key}")
+
+
+def archive_bag_message(bag_url):
+    """
+    Generates bag archive messages.
+    """
+    url = urlparse(bag_url)
+    if url.scheme == 's3':
+        bucket = url.netloc
+        key = url.path.lstrip('/')
+        return {
+            'namespace': bucket,
+            'key': key
+        }
+    else:
+        raise ValueError(f"Unrecognised url scheme: {bag_url}")
+
+
+@log_on_error
+def handler(event, _ctx=None, sns_client=None):
+    logger.info(f"received {event}")
+
+    ensure_valid_request(event)
+
+    bag_url = event['bagURL']
+
+    message = archive_bag_message(bag_url)
+    logger.debug(f"sns-message: {message}")
+
+    topic_arn = os.environ['INGEST_TOPIC_ARN']
```

Comment:
nit: I prefer to extract environment variables at the top of a function, so if there's a config error we fail immediately.